<div class="container">

<table style="width: 100%;"><tr>
<td>query_pancan_value</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Query Single Identifier or Signature Value from Pan-cancer Database</h2>

<h3>Description</h3>

<p>Query Single Identifier or Signature Value from Pan-cancer Database
</p>


<h3>Usage</h3>

<pre><code class="language-R">query_pancan_value(
  molecule,
  data_type = c("mRNA", "transcript", "protein", "mutation", "cnv", "methylation",
    "miRNA", "fusion", "promoter", "APOBEC"),
  database = c("toil", "ccle", "pcawg"),
  reset_id = NULL,
  opt_pancan = .opt_pancan
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>molecule</code></td>
<td>
<p>a molecular identifier (e.g., "TP53") or a formula specifying
genomic signature (<code>"TP53 + 2 * KRAS - 1.3 * PTEN"</code>).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>data_type</code></td>
<td>
<p>data type. Can be one of "mRNA", "transcript", "protein",
"mutation", "cnv", "methylation", "miRNA".</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>database</code></td>
<td>
<p>database, either 'toil' for TCGA TARGET GTEx, or 'ccle' for
CCLE.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>reset_id</code></td>
<td>
<p>if not <code>NULL</code>, set the specified variable at parent frame to "Signature".</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>opt_pancan</code></td>
<td>
<p>other extra parameters passing to the underlying functions.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p><code>query_pancan_value()</code> provide convenient interface to download multi-omics
data from 3 databases by specifying one or several canonical datasets. It is
derived from <code>query_pancan_value()</code> and support query for genomic signature.
To query comprehensive datasets that UCSCXena supports, users can check
<code>UCSCXenaTools::XenaData</code> and use <code>get_pancan_value()</code> directly.
</p>
<p>Option <code>opt_pancan</code> is a nested list and allow to adjust the downloading details.
For now, only <code>cnv(toil)</code>,<code>methylation(toil)</code>,<code>miRNA(toil)</code>,<code>miRNA(pcawg)</code>,<code>promoter(pcawg)</code>
support optional parameters. The default set is <code>.opt_pancan</code> and we check meanings of sublist(parameters)
through the following relationship.
</p>


<h3>Value</h3>

<p>a list.
</p>


<h3>"toil" database</h3>


<ol>
<li>
<p> mRNA–<code>get_pancan_gene_value()</code>
</p>
</li>
<li>
<p> transcript–<code>get_pancan_transcript_value()</code>
</p>
</li>
<li>
<p> protein–<code>get_pancan_protein_value()</code>
</p>
</li>
<li>
<p> mutation–<code>get_pancan_mutation_status()</code>
</p>
</li>
<li>
<p> cnv–<code>get_pancan_cn_value()</code>
</p>
</li>
<li>
<p> methylation–<code>get_pancan_methylation_value()</code>
</p>
</li>
<li>
<p> miRNA–<code>get_pancan_miRNA_value()</code>
</p>
</li>
</ol>
<h3>"ccle" database</h3>


<ol>
<li>
<p> mRNA–<code>get_ccle_gene_value()</code>
</p>
</li>
<li>
<p> protein–<code>get_ccle_protein_value()</code>
</p>
</li>
<li>
<p> mutation–<code>get_ccle_mutation_status()</code>
</p>
</li>
<li>
<p> cnv–<code>get_ccle_cn_value()</code>
</p>
</li>
</ol>
<h3>"pcawg" database</h3>


<ol>
<li>
<p> mRNA–<code>get_pcawg_gene_value()</code>
</p>
</li>
<li>
<p> miRNA–<code>get_pcawg_miRNA_value()</code>
</p>
</li>
<li>
<p> promoter–<code>get_pcawg_promoter_value()</code>
</p>
</li>
<li>
<p> fusion–<code>get_pcawg_fusion_value()</code>
</p>
</li>
<li>
<p> APOBEC–<code>get_pcawg_APOBEC_mutagenesis_value()</code>
</p>
</li>
</ol>
<h3>Examples</h3>

<pre><code class="language-R">## Not run: 
query_pancan_value("KRAS")
query_pancan_value("KRAS", database = "ccle")
query_pancan_value("KRAS", database = "pcawg")


query_pancan_value("ENSG00000000419",
  database = "pcawg",
  data_type = "fusion"
) # gene symbol also work

.opt_pancan

opt_pancan = list(toil_cnv = list(use_thresholded_data = FALSE))
query_pancan_value("PTEN",data_type = "cnv", database = "toil", opt_pancan = opt_pancan)


opt_pancan = list(toil_methylation = list(type = "450K",rule_out = "cg21115430", aggr = "Q25"))
query_pancan_value("PTEN",data_type = "methylation", database = "toil", opt_pancan = opt_pancan)


## End(Not run)
</code></pre>


</div>