<div class="container">

<table style="width: 100%;"><tr>
<td>vis_toil_TvsN</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Visualize Pan-cancer TPM (tumor (TCGA) vs Normal (TCGA &amp; GTEx))</h2>

<h3>Description</h3>

<p>Visualize Pan-cancer TPM (tumor (TCGA) vs Normal (TCGA &amp; GTEx))
</p>


<h3>Usage</h3>

<pre><code class="language-R">vis_toil_TvsN(
  Gene = "TP53",
  Mode = c("Boxplot", "Violinplot"),
  data_type = "mRNA",
  Show.P.value = TRUE,
  Show.P.label = TRUE,
  Method = c("wilcox.test", "t.test"),
  values = c("#DF2020", "#DDDF21"),
  TCGA.only = FALSE,
  draw_quantiles = c(0.25, 0.5, 0.75),
  trim = TRUE,
  include.Tumor.only = FALSE,
  opt_pancan = .opt_pancan
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>Gene</code></td>
<td>
<p>a molecular identifier (e.g., "TP53") or a formula specifying
genomic signature (<code>"TP53 + 2 * KRAS - 1.3 * PTEN"</code>).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Mode</code></td>
<td>
<p>"Boxplot" or "Violinplot" to represent data</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>data_type</code></td>
<td>
<p>choose gene profile type,
including "mRNA", "transcript", "protein", "mutation", "cnv", "methylation", "miRNA".</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Show.P.value</code></td>
<td>
<p><code>TRUE</code> or <code>FALSE</code> whether to count P value</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Show.P.label</code></td>
<td>
<p><code>TRUE</code> or <code>FALSE</code> present p value with number or label <code>*</code>, <code style="white-space: pre;">⁠**⁠</code>, <code style="white-space: pre;">⁠***⁠</code> and <code style="white-space: pre;">⁠****⁠</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Method</code></td>
<td>
<p>default method is wilcox.test</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>values</code></td>
<td>
<p>the color to fill tumor or normal</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>TCGA.only</code></td>
<td>
<p>include samples only from TCGA dataset</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>draw_quantiles</code></td>
<td>
<p>draw quantiles for violinplot</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>trim</code></td>
<td>
<p>whether trim the violin</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>include.Tumor.only</code></td>
<td>
<p>if <code>TRUE</code>, include "UVM" and "MESO" these two types
with matched normals samples.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>opt_pancan</code></td>
<td>
<p>specify one dataset for some molercular profiles</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>a <code>ggplot</code> object
</p>


<h3>Examples</h3>

<pre><code class="language-R">## Not run: 
p &lt;- vis_toil_TvsN(Gene = "TP53", Mode = "Violinplot", Show.P.value = FALSE, Show.P.label = FALSE)
p &lt;- vis_toil_TvsN(Gene = "TP53", Mode = "Boxplot", Show.P.value = FALSE, Show.P.label = FALSE)

## End(Not run)
</code></pre>


</div>