<div class="container">

<table style="width: 100%;"><tr>
<td>getTCGAdata</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Get TCGA Common Data Sets by Project ID and Property</h2>

<h3>Description</h3>

<p>This is the most useful function for user to download common
TCGA datasets, it is similar to <code>getFirehoseData</code> function in <code>RTCGAToolbox</code>
package.
</p>


<h3>Usage</h3>

<pre><code class="language-R">getTCGAdata(
  project = NULL,
  clinical = TRUE,
  download = FALSE,
  forceDownload = FALSE,
  destdir = tempdir(),
  mRNASeq = FALSE,
  mRNAArray = FALSE,
  mRNASeqType = "normalized",
  miRNASeq = FALSE,
  exonRNASeq = FALSE,
  RPPAArray = FALSE,
  ReplicateBaseNormalization = FALSE,
  Methylation = FALSE,
  MethylationType = c("27K", "450K"),
  GeneMutation = FALSE,
  SomaticMutation = FALSE,
  GisticCopyNumber = FALSE,
  Gistic2Threshold = TRUE,
  CopyNumberSegment = FALSE,
  RemoveGermlineCNV = TRUE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>project</code></td>
<td>
<p>default is <code>NULL</code>. Should be one or more of TCGA project id (character vector) provided by Xena.
See all available project id, please use <code>availTCGA("ProjectID")</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>clinical</code></td>
<td>
<p>logical. if <code>TRUE</code>, download clinical information. Default is <code>TRUE</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>download</code></td>
<td>
<p>logical. if <code>TRUE</code>, download data, otherwise return a result list include data
information. Default is <code>FALSE</code>. You can set this to <code>FALSE</code> if you want to check what you will download or
use other function provided by <code>UCSCXenaTools</code> to filter result datasets you want to download.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>forceDownload</code></td>
<td>
<p>logical. if <code>TRUE</code>, force to download files no matter if exist. Default is <code>FALSE</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>destdir</code></td>
<td>
<p>specify a location to store download data. Default is system temp directory.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>mRNASeq</code></td>
<td>
<p>logical. if <code>TRUE</code>, download mRNASeq data. Default is <code>FALSE</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>mRNAArray</code></td>
<td>
<p>logical. if <code>TRUE</code>, download mRNA microarray data. Default is <code>FALSE</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>mRNASeqType</code></td>
<td>
<p>character vector. Can be one, two or three
in <code>c("normalized", "pancan normalized", "percentile")</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>miRNASeq</code></td>
<td>
<p>logical. if <code>TRUE</code>, download miRNASeq data. Default is <code>FALSE</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>exonRNASeq</code></td>
<td>
<p>logical. if <code>TRUE</code>, download exon RNASeq data. Default is <code>FALSE</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>RPPAArray</code></td>
<td>
<p>logical. if <code>TRUE</code>, download RPPA data. Default is <code>FALSE</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ReplicateBaseNormalization</code></td>
<td>
<p>logical. if <code>TRUE</code>, download RPPA data by Replicate Base
Normalization (RBN). Default is <code>FALSE</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Methylation</code></td>
<td>
<p>logical. if <code>TRUE</code>, download DNA Methylation data. Default is <code>FALSE</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>MethylationType</code></td>
<td>
<p>character vector. Can be one or two in <code>c("27K", "450K")</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>GeneMutation</code></td>
<td>
<p>logical. if <code>TRUE</code>, download gene mutation data. Default is <code>FALSE</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>SomaticMutation</code></td>
<td>
<p>logical. if <code>TRUE</code>, download somatic mutation data. Default is <code>FALSE</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>GisticCopyNumber</code></td>
<td>
<p>logical. if <code>TRUE</code>, download Gistic2 Copy Number data. Default is <code>FALSE</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Gistic2Threshold</code></td>
<td>
<p>logical. if <code>TRUE</code>, download Threshold Gistic2 data. Default is <code>TRUE</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>CopyNumberSegment</code></td>
<td>
<p>logical. if <code>TRUE</code>, download Copy Number Segment data. Default is <code>FALSE</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>RemoveGermlineCNV</code></td>
<td>
<p>logical. if <code>TRUE</code>, download Copy Number Segment data which has removed
germline copy number variation. Default is <code>TRUE</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p>other argument to <code>download.file</code> function</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>TCGA Common Data Sets are frequently used for biological analysis.
To make easier to achieve these data, this function provide really easy
options to choose datasets and behavior. All availble information about
datasets of TCGA can access vis <code>availTCGA()</code> and check with <code>showTCGA()</code>.
</p>


<h3>Value</h3>

<p>if <code>download=TRUE</code>, return <code>data.frame</code> from <code>XenaDownload</code>,
otherwise return a list including <code>XenaHub</code> object and datasets information
</p>


<h3>Author(s)</h3>

<p>Shixiang Wang <a href="mailto:w_shixiang@163.com">w_shixiang@163.com</a>
</p>


<h3>Examples</h3>

<pre><code class="language-R">###### get data, but not download

# 1 choose project and data types you wanna download
getTCGAdata(project = "LUAD", mRNASeq = TRUE, mRNAArray = TRUE,
mRNASeqType = "normalized", miRNASeq = TRUE, exonRNASeq = TRUE,
RPPAArray = TRUE, Methylation = TRUE, MethylationType = "450K",
GeneMutation = TRUE, SomaticMutation = TRUE)

# 2 only choose 'LUAD' and its clinical data
getTCGAdata(project = "LUAD")
## Not run: 
###### download datasets

# 3 download clinical datasets of LUAD and LUSC
getTCGAdata(project = c("LUAD", "LUSC"), clinical = TRUE, download = TRUE)

# 4 download clinical, RPPA and gene mutation datasets of LUAD and LUSC
# getTCGAdata(project = c("LUAD", "LUSC"), clinical = TRUE, RPPAArray = TRUE, GeneMutation = TRUE)

## End(Not run)
</code></pre>


</div>