<div class="container">

<table style="width: 100%;"><tr>
<td>vis_gene_immune_cor</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Heatmap for Correlation between Gene and Immune Signatures</h2>

<h3>Description</h3>

<p>Heatmap for Correlation between Gene and Immune Signatures
</p>


<h3>Usage</h3>

<pre><code class="language-R">vis_gene_immune_cor(
  Gene = "TP53",
  cor_method = "spearman",
  data_type = "mRNA",
  Immune_sig_type = "Cibersort",
  Plot = "TRUE",
  opt_pancan = .opt_pancan
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>Gene</code></td>
<td>
<p>a molecular identifier (e.g., "TP53") or a formula specifying
genomic signature (<code>"TP53 + 2 * KRAS - 1.3 * PTEN"</code>).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>cor_method</code></td>
<td>
<p>correlation method</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>data_type</code></td>
<td>
<p>choose gene profile type,
including "mRNA", "transcript", "protein", "mutation", "cnv", "methylation", "miRNA".</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Immune_sig_type</code></td>
<td>
<p>quantification method, default is "Cibersort"</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Plot</code></td>
<td>
<p>output the plot directly, default 'TRUE'</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>opt_pancan</code></td>
<td>
<p>specify one dataset for some molercular profiles</p>
</td>
</tr>
</table>
<h3>Examples</h3>

<pre><code class="language-R">## Not run: 
p &lt;- vis_gene_immune_cor(Gene = "TP53")

## End(Not run)
</code></pre>


</div>