<div class="container">

<table style="width: 100%;"><tr>
<td>nice_tidy</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>V4-V5 16S rRNA gene amplicons, clean OTU table in tidy/long format (N-ICE, 2015)</h2>

<h3>Description</h3>

<p>Original OTU table (nice) in "long" format.
</p>


<h3>Usage</h3>

<pre><code class="language-R">nice_tidy
</code></pre>


<h3>Format</h3>



<h4><code>nice_tidy</code></h4>

<p>A data frame with 4716 rows and 10 columns:
</p>

<dl>
<dt>Sample</dt>
<dd>
<p>Sample ID</p>
</dd>
<dt>Abundance</dt>
<dd>
<p>Abundance</p>
</dd>
<dt>OTU</dt>
<dd>
<p>OTU ID</p>
</dd>
<dt>Domain</dt>
<dd>
<p>Domain level classification of OTU</p>
</dd>
<dt>Phylum</dt>
<dd>
<p>Domain level classification of OTU</p>
</dd>
<dt>Class</dt>
<dd>
<p>Domain level classification of OTU</p>
</dd>
<dt>Order</dt>
<dd>
<p>Domain level classification of OTU</p>
</dd>
<dt>Family</dt>
<dd>
<p>Domain level classification of OTU</p>
</dd>
<dt>Genus</dt>
<dd>
<p>Domain level classification of OTU</p>
</dd>
<dt>Species</dt>
<dd>
<p>Domain level classification of OTU</p>
</dd>
</dl>
<p>...

</p>



<h3>Details</h3>

<p>A new column (Sample) includes the sample identifiers and a new column
(Abundance) includes the abundance for each OTU.
For details on OTU table processing see help pages for nice and nice_raw.
</p>
<p>Some details on N-ICE dataset:
</p>
<p>This dataset resulted from the Norwegian Young Sea Ice expedition (N-ICE)
in 2015 (Granskog et al., 2018). The sample processing and DNA sequencing
were described in de Sousa et al., 2019, the bioinformatic processing was
performed by the MGnify platform (v5) (Mitchell et al., 2020).
</p>
<p>Since the purpose of this dataset if for creating examples and testing the package,
we did not apply strict quality control to the final OTU table. Thus, we
didn't remove singletons, etc. However, we did remove any
non-prokarotic OTUs and organelles, if any (Pascoal et al., 2022).
</p>


<h3>Source</h3>

<p><a href="https://www.ebi.ac.uk/metagenomics/studies/MGYS00001922#analysis">https://www.ebi.ac.uk/metagenomics/studies/MGYS00001922#analysis</a>
</p>


<h3>References</h3>


<ul>
<li>
<p> Mitchell, A. L., Almeida, A., Beracochea, M., Boland, M., Burgin, J., Cochrane, G., Crusoe, M. R., Kale, V., Potter, S. C., Richardson, L. J., Sakharova, E., Scheremetjew, M., Korobeynikov, A., Shlemov, A., Kunyavskaya, O., Lapidus, A., &amp; Finn, R. D. (2019). MGnify: the microbiome analysis resource in 2020. Nucleic Acids Research, 48(D1), D570–D578.
</p>
</li>
<li>
<p> Granskog, M. A., Fer, I., Rinke, A., &amp; Steen, H. (2018). Atmosphere-Ice-Ocean-Ecosystem Processes in a Thinner Arctic Sea Ice Regime: The Norwegian Young Sea ICE (N-ICE2015) Expedition. Journal of Geophysical Research: Oceans, 123(3), 1586–1594.
</p>
</li>
<li>
<p> de Sousa, A. G. G., Tomasino, M. P., Duarte, P., Fernández-Méndez, M., Assmy, P., Ribeiro, H., Surkont, J., Leite, R. B., Pereira-Leal, J. B., Torgo, L., &amp; Magalhães, C. (2019). Diversity and Composition of Pelagic Prokaryotic and Protist Communities in a Thin Arctic Sea-Ice Regime. Microbial Ecology, 78(2), 388–408.
</p>
</li>
<li>
<p> Pascoal, F., Costa, R., Assmy, P., Duarte, P., &amp; Magalhães, C. (2022). Exploration of the Types of Rarity in the Arctic Ocean from the Perspective of Multiple Methodologies. Microbial Ecology, 84(1), 59–72.
</p>
</li>
</ul>
<h3>See Also</h3>

<p><code>nice()</code>, nice_raw, nice_env
</p>


</div>