<div class="container">

<table style="width: 100%;"><tr>
<td>CancerEngine-class</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>The "CancerEngine" Class</h2>

<h3>Description</h3>

<p>A CancerEngine combines a CancerModel (which defines the combinatorics
of hits that produce cancer subtypes) with a pair of gene expression
Engines that can be used to simulate microarray data depending on the
presence or absence of different hits.
</p>


<h3>Usage</h3>

<pre><code class="language-R">CancerEngine(cm, base, altered)
## S4 method for signature 'CancerEngine'
summary(object, ...)
## S4 method for signature 'CancerEngine'
nrow(x)
## S4 method for signature 'CancerEngine'
nComponents(object, ...)
## S4 method for signature 'CancerEngine'
rand(object, n, ...)
ClinicalEngine(nFeatures, nClusters, isWeighted,
               bHyp = NULL, survivalModel = NULL,
               SURV = function(n) rnorm(n, 0, 0.3),
               OUT = function(n) rnorm(n, 0, 0.3))
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>cm</code></td>
<td>
<p>object of class <code>CancerModel</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>base</code></td>
<td>
<p>character string giving the name of an <code>Engine</code>
or <code>EngineWithActivity</code>, or an object of class <code>Engine</code>.
Represents the expected gene expression in the absence of any hits.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>altered</code></td>
<td>
<p>character string giving the name of an <code>Engine</code>
or <code>EngineWithActivity</code>, or an object of class <code>Engine</code>.
Represents the expected gene expression in the presence of hits.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>object</code></td>
<td>
<p>object of class <code>CancerEngine</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>x</code></td>
<td>
<p>object of class <code>CancerEngine</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>n</code></td>
<td>
<p>numeric scalar representing number of samples to be simulated</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p>extra arguments for generic routines</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>nFeatures</code></td>
<td>
<p>an integer; the number of simulated clinical features</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>nClusters</code></td>
<td>
<p>an integer; the number of simulated clusters or subtypes</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>isWeighted</code></td>
<td>
<p>a logical value; used to determine whether the prevalence
of subtypes is equal (unweighted) or unequal (weighted).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>bHyp</code></td>
<td>
<p>an object of the class <code>BlockHyperParameters</code>. If NULL,
then it is constructed using the default clinical parameters. See the
Values section below.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>survivalModel</code></td>
<td>
<p>an object of the <code>SurvivalModel</code> class.
If NULL, then it is constructed using the default parameters.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>SURV</code></td>
<td>
<p>a function; the same as used in a <code>CancerModel</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>OUT</code></td>
<td>
<p>a function; the same as used in a <code>CancerModel</code>.</p>
</td>
</tr>
</table>
<h3>Objects from the Class</h3>

<p>Although objects of the class can be created by a direct call to
new, the preferred method is to use the
<code>CancerEngine</code> or <code>ClinicalEngine</code> generator functions.
</p>


<h3>Slots</h3>


<dl>
<dt>
<code>cm</code>:</dt>
<dd>
<p>A <code>CancerModel</code> object.</p>
</dd>
<dt>
<code>base</code>:</dt>
<dd>
<p>Object of class "character" giving the name of
an <code>Engine</code> or <code>EngineWithActivity</code>.
Represents the expected gene expression in the absence of any hits.</p>
</dd>
<dt>
<code>altered</code>:</dt>
<dd>
<p>Object of class "character" giving the name of
an <code>Engine</code> or <code>EngineWithActivity</code>.
Represents the expected gene expression in the presence of hits.</p>
</dd>
<dt>
<code>localenv</code>:</dt>
<dd>
<p>Object of class <code>"environment"</code>; used
in the internal implementation. </p>
</dd>
</dl>
<h3>Methods</h3>


<dl>
<dt>rand</dt>
<dd>
<p><code>signature(object = "CancerEngine")</code>: returns a list
containing two named components, <code>clinical</code> and <code>data</code>, The
clinical element is a data frame generated from the underlying
<code>CancerModel</code>, and the data element is a matrix generated by the
gene expression engines, altered by the appropriate "hits" present in
each simulated individual.</p>
</dd>
<dt>summary</dt>
<dd>
<p><code>signature(object = "CancerEngine")</code>: print
summaries of the underlying cancer models and gene expression engines
in the cancer engine. </p>
</dd>  </dl>
<h3>Values</h3>

<p>Both the <code>CancerEngine</code> and <code>ClinicalEngine</code>
constructors return objects of the <code>CancerEngine</code> class. The only
practical differences are the default parameters set by the constructors.
The primary change is in the <code>CancerModel</code> slot. The
<code>CancerEngine</code> expects you to construct your own <code>CancerModel</code>
explicitly before producing a <code>CancerEngine</code>.
</p>
<p>By contrast, the <code>ClinicalEngine</code> expects to use many fewer
features, so gives a simpler interface and uses its parameters to construct the
<code>CancerModel</code> for you. The "HIT" function will use 2 hits per subtype
when there are fewer than 15 features, 3 hits between 15 and 45 features,
and the older default of 5 hits when there are more than 45 features. The
total number of possible hits is set equal to the number of features (N) when
there are fewer than 12 features, approximately N/3 up to 50 features,
approximately N/5 up to 100, and is then locked at 20. If <code>isWeighted</code>
is <code>TRUE</code>, subtype prevalences are chosen from a Dirichlet
distribution. Otherwise, each subtype is equally likely.
</p>
<p>The <code>nrow</code> and <code>nComponents</code> methods both return
non-negative integer values describing the number of rows (features)
and the number of components of the underlying gene expression or
clinical data Engines. The <code>rand</code> method returns a matrix with
<code>n</code> columns of smiulated data.
</p>


<h3>Author(s)</h3>

<p>Kevin R. Coombes <a href="mailto:krc@silicovore.com">krc@silicovore.com</a>,
Caitlin E. Coombes <a href="mailto:caitlin.coombes@osumc.edu">caitlin.coombes@osumc.edu</a>
</p>


<h3>References</h3>

<p>Zhang J, Coombes KR.<br><em>Sources of variation in false discovery rate estimation include
sample size, correlation, and inherent differences between groups.</em><br>
BMC Bioinformatics. 2012; 13 Suppl 13:S1. 
</p>


<h3>See Also</h3>

<p><code>CancerModel</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">showClass("CancerEngine")
set.seed(391629)
## Set up survival outcome; baseline is exponential
sm &lt;- SurvivalModel(baseHazard=1/5, accrual=5, followUp=1)
## Build a CancerModel with 6 subtypes
nBlocks &lt;- 20    # number of possible hits
cm &lt;- CancerModel(name="cansim",
                  nPossible=nBlocks,
                  nPattern=6,
                  OUT = function(n) rnorm(n, 0, 1), 
                  SURV= function(n) rnorm(n, 0, 1),
                  survivalModel=sm)
## Include 100 blocks/pathways that are not hit by cancer
nTotalBlocks &lt;- nBlocks + 100
## Assign values to hyperparameters
## block size
blockSize &lt;- round(rnorm(nTotalBlocks, 100, 30))
## log normal mean hypers
mu0    &lt;- 6
sigma0 &lt;- 1.5
## log normal sigma hypers
rate   &lt;- 28.11
shape  &lt;- 44.25
## block corr
p &lt;- 0.6
w &lt;- 5
## Set up the baseline Engine
rho &lt;- rbeta(nTotalBlocks, p*w, (1-p)*w)
base &lt;- lapply(1:nTotalBlocks,
               function(i) {
                 bs &lt;- blockSize[i]
                 co &lt;- matrix(rho[i], nrow=bs, ncol=bs)
                 diag(co) &lt;- 1
                 mu &lt;- rnorm(bs, mu0, sigma0)
                 sigma &lt;- matrix(1/rgamma(bs, rate=rate, shape=shape), nrow=1)
                 covo &lt;- co *(t(sigma) %*% sigma)
                 MVN(mu, covo)
               })
eng &lt;- Engine(base)
## Alter the means if there is a hit
altered &lt;- alterMean(eng, normalOffset, delta=0, sigma=1)
## Build the CancerEngine using character strings
object &lt;- CancerEngine(cm, "eng", "altered")
## Or build it using the actual Engine components
ob &lt;- CancerEngine(cm, eng, altered)
summary(object)
summary(ob)
## Simulate the data
dset &lt;- rand(object, 20)
summary(dset$clinical)
summary(dset$data[, 1:3])
</code></pre>


</div>