<div class="container">

<table style="width: 100%;"><tr>
<td>rflexdog</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Simulate GBS data from the <code>flexdog</code> likelihood.</h2>

<h3>Description</h3>

<p>This will take a vector of genotypes and a vector of total read-counts, then generate a vector of reference
counts. To get the genotypes, you could use <code>rgeno</code>. The likelihood used to generate read-counts
is described in detail in Gerard et. al. (2018).
</p>


<h3>Usage</h3>

<pre><code class="language-R">rflexdog(sizevec, geno, ploidy, seq = 0.005, bias = 1, od = 0.001)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>sizevec</code></td>
<td>
<p>A vector of total read-counts for the individuals.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>geno</code></td>
<td>
<p>A vector of genotypes for the individuals. I.e. the number of reference alleles
each individual has.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ploidy</code></td>
<td>
<p>The ploidy of the species.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>seq</code></td>
<td>
<p>The sequencing error rate.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>bias</code></td>
<td>
<p>The bias parameter. Pr(a read after selected) / Pr(A read after selected).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>od</code></td>
<td>
<p>The overdispersion parameter. See the
Details of the <code>rho</code> variable in <code>betabinom</code>.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>A vector the same length as <code>sizevec</code>. The ith element
is the number of reference counts for individual i.
</p>


<h3>Author(s)</h3>

<p>David Gerard
</p>


<h3>References</h3>


<ul>
<li>
<p>Gerard, D., Ferrão, L. F. V., Garcia, A. A. F., &amp; Stephens, M. (2018). Genotyping Polyploids from Messy Sequencing Data. <em>Genetics</em>, 210(3), 789-807. <a href="https://doi.org/10.1534/genetics.118.301468">doi:10.1534/genetics.118.301468</a>.
</p>
</li>
<li>
<p>Gerard, David, and Luís Felipe Ventorim Ferrão. "Priors for genotyping polyploids." Bioinformatics 36, no. 6 (2020): 1795-1800. <a href="https://doi.org/10.1093/bioinformatics/btz852">doi:10.1093/bioinformatics/btz852</a>.
</p>
</li>
</ul>
<h3>See Also</h3>

<p><code>rgeno</code> for a way to generate genotypes of individuals. <code>rbetabinom</code>
for how we generate the read-counts.
</p>


<h3>Examples</h3>

<pre><code class="language-R">set.seed(1)
n       &lt;- 100
ploidy  &lt;- 6

## Generate the genotypes of individuals from an F1 population,
## where the first parent has 1 copy of the reference allele
## and the second parent has two copies of the reference
## allele.
genovec &lt;- rgeno(n = n, ploidy = ploidy, model = "f1",
                 p1geno = 1, p2geno = 2)

## Get the total number of read-counts for each individual.
## Ideally, you would take this from real data as the total
## read-counts are definitely not Poisson.
sizevec &lt;- stats::rpois(n = n, lambda = 200)

## Generate the counts of reads with the reference allele
## when there is a strong bias for the reference allele
## and there is no overdispersion.
refvec  &lt;- rflexdog(sizevec = sizevec, geno = genovec,
                    ploidy = ploidy, seq = 0.001,
                    bias = 0.5, od = 0)

## Plot the simulated data using plot_geno.
plot_geno(refvec = refvec, sizevec = sizevec,
          ploidy = ploidy, seq = 0.001, bias = 0.5)

</code></pre>


</div>