<div class="container">

<table style="width: 100%;"><tr>
<td>snpdat</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>GBS data from Shirasawa et al (2017)</h2>

<h3>Description</h3>

<p>Contains counts of reference alleles and total read counts from the GBS data of Shirasawa et al (2017) for
the three SNPs used as examples in Gerard et. al. (2018).
</p>


<h3>Usage</h3>

<pre><code class="language-R">snpdat
</code></pre>


<h3>Format</h3>

<p>A <code>tibble</code> with 419 rows and 4 columns:
</p>

<dl>
<dt>id</dt>
<dd>
<p>The identification label of the individuals.</p>
</dd>
<dt>snp</dt>
<dd>
<p>The SNP label.</p>
</dd>
<dt>counts</dt>
<dd>
<p>The number of read-counts that support the reference allele.</p>
</dd>
<dt>size</dt>
<dd>
<p>The total number of read-counts at a given SNP.</p>
</dd>
</dl>
<h3>Value</h3>

<p>A <code>tibble</code>. See the Format Section.
</p>


<h3>Source</h3>

<p><a href="https://doi.org/10.1038/srep44207">doi:10.1038/srep44207</a>
</p>


<h3>References</h3>


<ul>
<li>
<p>Shirasawa, Kenta, Masaru Tanaka, Yasuhiro Takahata, Daifu Ma, Qinghe Cao, Qingchang Liu, Hong Zhai, Sang-Soo Kwak, Jae Cheol Jeong, Ung-Han Yoon, Hyeong-Un Lee, Hideki Hirakawa, and Sahiko Isobe "A high-density SNP genetic map consisting of a complete set of homologous groups in autohexaploid sweetpotato (Ipomoea batatas)." <em>Scientific Reports 7</em> (2017). <a href="https://doi.org/10.1038/srep44207">doi:10.1038/srep44207</a>
</p>
</li>
<li>
<p>Gerard, D., Ferr√£o, L. F. V., Garcia, A. A. F., &amp; Stephens, M. (2018). Genotyping Polyploids from Messy Sequencing Data. <em>Genetics</em>, 210(3), 789-807. <a href="https://doi.org/10.1534/genetics.118.301468">doi:10.1534/genetics.118.301468</a>.
</p>
</li>
</ul>
</div>