<div class="container">

<table style="width: 100%;"><tr>
<td>plot_geno</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Make a genotype plot.</h2>

<h3>Description</h3>

<p>The x-axis is the counts of the non-reference allele,
and the y-axis is the counts of the reference allele.
Transparency is controlled by the <code>maxpostprob</code>
vector. These types of plots are used in Gerard et. al. (2018) and
Gerard and Ferrão (2020).
</p>


<h3>Usage</h3>

<pre><code class="language-R">plot_geno(
  refvec,
  sizevec,
  ploidy,
  p1ref = NULL,
  p1size = NULL,
  p2ref = NULL,
  p2size = NULL,
  geno = NULL,
  seq = 0,
  bias = 1,
  maxpostprob = NULL,
  p1geno = NULL,
  p2geno = NULL,
  use_colorblind = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>refvec</code></td>
<td>
<p>A vector of non-negative integers. The number of
reference reads observed in the individuals</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>sizevec</code></td>
<td>
<p>A vector of positive integers. The total number of
reads in the individuals.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ploidy</code></td>
<td>
<p>A non-negative integer. The ploidy of the species.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>p1ref</code></td>
<td>
<p>A vector of non-negative integers. The number of
reference reads observed in parent 1 (if the individuals are
all siblings).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>p1size</code></td>
<td>
<p>A vector of positive integers. The total number of
reads in parent 1 (if the individuals are
all siblings).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>p2ref</code></td>
<td>
<p>A vector of non-negative integers. The number of
reference reads observed in parent 2 (if the individuals are
all siblings).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>p2size</code></td>
<td>
<p>A vector of positive integers. The total number of
reads in parent 2 (if the individuals are
all siblings).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>geno</code></td>
<td>
<p>The individual genotypes.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>seq</code></td>
<td>
<p>The sequencing error rate.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>bias</code></td>
<td>
<p>The bias parameter.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>maxpostprob</code></td>
<td>
<p>A vector of the posterior probabilities of being at
the modal genotype.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>p1geno</code></td>
<td>
<p>Parent 1's genotype.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>p2geno</code></td>
<td>
<p>Parent 2's genotype.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>use_colorblind</code></td>
<td>
<p>A logical. Should we use a colorblind safe palette (<code>TRUE</code>),
or not (<code>FALSE</code>)? Only allowed if <code>ploidy &lt;= 6</code>.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>If parental genotypes are provided (<code>p1geno</code> and <code>p2geno</code>) then
they will be colored the same as the offspring. Since they are often hard to see,
a small black dot will also indicate their position.
</p>


<h3>Value</h3>

<p>A <code>ggplot</code> object for the genotype plot.
</p>


<h3>Author(s)</h3>

<p>David Gerard
</p>


<h3>References</h3>


<ul>
<li>
<p>Gerard, D., Ferrão, L. F. V., Garcia, A. A. F., &amp; Stephens, M. (2018). Genotyping Polyploids from Messy Sequencing Data. <em>Genetics</em>, 210(3), 789-807. <a href="https://doi.org/10.1534/genetics.118.301468">doi:10.1534/genetics.118.301468</a>.
</p>
</li>
<li>
<p>Gerard, David, and Luís Felipe Ventorim Ferrão. "Priors for genotyping polyploids." Bioinformatics 36, no. 6 (2020): 1795-1800. <a href="https://doi.org/10.1093/bioinformatics/btz852">doi:10.1093/bioinformatics/btz852</a>.
</p>
</li>
</ul>
<h3>Examples</h3>

<pre><code class="language-R">data("snpdat")
refvec  &lt;- snpdat$counts[snpdat$snp == "SNP1"]
sizevec &lt;- snpdat$size[snpdat$snp == "SNP1"]
ploidy  &lt;- 6
plot_geno(refvec = refvec, sizevec = sizevec, ploidy = ploidy)

</code></pre>


</div>