<div class="container">

<table style="width: 100%;"><tr>
<td>rgeno</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Simulate individual genotypes from one of the supported <code>flexdog</code> models.</h2>

<h3>Description</h3>

<p>This will simulate genotypes of a sample of individuals drawn from one of the populations supported by
<code>flexdog</code>. See the details of <code>flexdog</code> for the models allowed. These genotype
distributions are described in detail in Gerard and Ferrão (2020).
</p>


<h3>Usage</h3>

<pre><code class="language-R">rgeno(
  n,
  ploidy,
  model = c("hw", "bb", "norm", "f1", "s1", "flex", "uniform"),
  allele_freq = NULL,
  od = NULL,
  p1geno = NULL,
  p2geno = NULL,
  pivec = NULL,
  mu = NULL,
  sigma = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>n</code></td>
<td>
<p>The number of observations.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ploidy</code></td>
<td>
<p>The ploidy of the species.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>model</code></td>
<td>
<p>What form should the prior take? See Details in <code>flexdog</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>allele_freq</code></td>
<td>
<p>If <code>model = "hw"</code>, then this is the allele frequency of the population.
For any other model, this should be <code>NULL</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>od</code></td>
<td>
<p>If <code>model = "bb"</code>, then this is the overdispersion parameter of the beta-binomial
distribution. See <code>betabinom</code> for details. For any other model, this should be
<code>NULL</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>p1geno</code></td>
<td>
<p>Either the first parent's genotype if <code>model = "f1"</code>,
or the only parent's
genotype if <code>model = "s1"</code>.
For any other model, this should be <code>NULL</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>p2geno</code></td>
<td>
<p>The second parent's genotype if <code>model = "f1"</code>.
For any other model, this should be <code>NULL</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pivec</code></td>
<td>
<p>A vector of probabilities. If <code>model = "ash"</code>, then this represents
the mixing proportions of the discrete uniforms. If
<code>model = "flex"</code>, then element <code>i</code> is the probability of genotype <code>i - 1</code>.
For any other model, this should be <code>NULL</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>mu</code></td>
<td>
<p>If <code>model = "norm"</code>, this is the mean of the normal.
For any other model, this should be <code>NULL</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>sigma</code></td>
<td>
<p>If <code>model = "norm"</code>, this is the standard deviation of the normal.
For any other model, this should be <code>NULL</code>.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>List of non-<code>NULL</code> arguments:
</p>

<dl>
<dt>
<code>model = "flex"</code>:</dt>
<dd>
<p><code>pivec</code></p>
</dd>
<dt>
<code>model = "hw"</code>:</dt>
<dd>
<p><code>allele_freq</code></p>
</dd>
<dt>
<code>model = "f1"</code>:</dt>
<dd>
<p><code>p1geno</code> and <code>p2geno</code></p>
</dd>
<dt>
<code>model = "s1"</code>:</dt>
<dd>
<p><code>p1geno</code></p>
</dd>
<dt>
<code>model = "uniform"</code>:</dt>
<dd>
<p>no non-<code>NULL</code> arguments</p>
</dd>
<dt>
<code>model = "bb"</code>:</dt>
<dd>
<p><code>allele_freq</code> and <code>od</code></p>
</dd>
<dt>
<code>model == "norm"</code>:</dt>
<dd>
<p><code>mu</code> and <code>sigma</code></p>
</dd>
</dl>
<h3>Value</h3>

<p>A vector of length <code>n</code> with the genotypes of the sampled individuals.
</p>


<h3>Author(s)</h3>

<p>David Gerard
</p>


<h3>References</h3>


<ul>
<li>
<p>Gerard, D., Ferrão, L. F. V., Garcia, A. A. F., &amp; Stephens, M. (2018). Genotyping Polyploids from Messy Sequencing Data. <em>Genetics</em>, 210(3), 789-807. <a href="https://doi.org/10.1534/genetics.118.301468">doi:10.1534/genetics.118.301468</a>.
</p>
</li>
<li>
<p>Gerard, David, and Luís Felipe Ventorim Ferrão. "Priors for genotyping polyploids." Bioinformatics 36, no. 6 (2020): 1795-1800. <a href="https://doi.org/10.1093/bioinformatics/btz852">doi:10.1093/bioinformatics/btz852</a>.
</p>
</li>
</ul>
<h3>Examples</h3>

<pre><code class="language-R">## F1 Population where parent 1 has 1 copy of the referenc allele
## and parent 2 has 4 copies of the reference allele.
ploidy &lt;- 6
rgeno(n = 10, ploidy = ploidy, model = "f1", p1geno = 1, p2geno = 4)

## A population in Hardy-Weinberge equilibrium with an
## allele frequency of 0.75
rgeno(n = 10, ploidy = ploidy, model = "hw", allele_freq = 0.75)

</code></pre>


</div>