<div class="container">

<table style="width: 100%;"><tr>
<td>oracle_joint</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>The joint probability of the genotype and the genotype estimate
of an oracle estimator.</h2>

<h3>Description</h3>

<p>This returns the joint distribution of the true genotypes and an oracle
estimator given perfect knowledge of the data generating process. This is a useful
approximation when you have a lot of individuals.
</p>


<h3>Usage</h3>

<pre><code class="language-R">oracle_joint(n, ploidy, seq, bias, od, dist)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>n</code></td>
<td>
<p>The read-depth.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ploidy</code></td>
<td>
<p>The ploidy of the individual.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>seq</code></td>
<td>
<p>The sequencing error rate.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>bias</code></td>
<td>
<p>The allele-bias.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>od</code></td>
<td>
<p>The overdispersion parameter.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>dist</code></td>
<td>
<p>The distribution of the alleles.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>To come up with <code>dist</code>, you need some additional assumptions.
For example, if the population is in Hardy-Weinberg equilibrium and
the allele frequency is <code>alpha</code> then you could calculate
<code>dist</code> using the R code: <code>dbinom(x = 0:ploidy, size = ploidy, prob = alpha)</code>.
Alternatively, if you know the genotypes of the individual's two parents are, say,
<code>ref_count1</code> and <code>ref_count2</code>, then you could use the <code>get_q_array</code>
function from the updog package: <code>get_q_array(ploidy)[ref_count1 + 1, ref_count2 + 1, ]</code>.
</p>
<p>See the Examples to see how to reconcile the output of <code>oracle_joint</code>
with <code>oracle_mis</code> and <code>oracle_mis_vec</code>.
</p>


<h3>Value</h3>

<p>A matrix. Element (i, j) is the joint probability of estimating
the genotype to be i+1 when the true genotype is j+1. That is, the
estimated genotype indexes the rows and the true genotype indexes
the columns. This is when
using an oracle estimator.
</p>


<h3>Author(s)</h3>

<p>David Gerard
</p>


<h3>References</h3>


<ul><li>
<p>Gerard, D., Ferr√£o, L. F. V., Garcia, A. A. F., &amp; Stephens, M. (2018). Genotyping Polyploids from Messy Sequencing Data. <em>Genetics</em>, 210(3), 789-807. <a href="https://doi.org/10.1534/genetics.118.301468">doi:10.1534/genetics.118.301468</a>.
</p>
</li></ul>
<h3>See Also</h3>


<dl>
<dt><code>oracle_plot</code></dt>
<dd>
<p>For visualizing the joint distribution output from <code>oracle_joint</code>.</p>
</dd>
<dt><code>oracle_mis_from_joint</code></dt>
<dd>
<p>For obtaining the same results as <code>oracle_mis</code>
directly from the output of <code>oracle_joint</code>.</p>
</dd>
<dt><code>oracle_mis_vec_from_joint</code></dt>
<dd>
<p>For obtaining the same results as <code>oracle_mis_vec</code>
directly from the output of <code>oracle_joint</code>.</p>
</dd>
<dt><code>oracle_cor_from_joint</code></dt>
<dd>
<p>For obtaining the same results as <code>oracle_cor</code>
directly from the output of <code>oracle_joint</code>.</p>
</dd>
</dl>
<h3>Examples</h3>

<pre><code class="language-R">## Hardy-Weinberg population with allele-frequency of 0.75.
## Moderate bias and moderate overdispersion.
ploidy &lt;- 4
dist &lt;- stats::dbinom(0:ploidy, ploidy, 0.75)
jd &lt;- oracle_joint(n = 100, ploidy = ploidy, seq = 0.001,
                   bias = 0.7, od = 0.01, dist = dist)
jd

## Get same output as oracle_mis this way:
1 - sum(diag(jd))
oracle_mis(n = 100, ploidy = ploidy, seq = 0.001,
           bias = 0.7, od = 0.01, dist = dist)

## Get same output as oracle_mis_vec this way:
1 - diag(sweep(x = jd, MARGIN = 2, STATS = colSums(jd), FUN = "/"))
oracle_mis_vec(n = 100, ploidy = ploidy, seq = 0.001,
               bias = 0.7, od = 0.01, dist = dist)

</code></pre>


</div>